staking/contracts/contracts.py

```diff
@@ -104,16 +115,20 @@ def calculate_rewards(addr: Expr) -> Expr:
         # Duration
         (duration := ScratchVar()).store(end.load() - start.load()),
 
-        # Calculate time since last updated
+        # Calculate rewards
         (rewards := ScratchVar()).store(
-            App.localGet(addr, Bytes("AS")) * duration.load() / Int(31557600) * App.globalGet(Bytes("FR")) / Int(10000)
+            App.localGet(addr, Bytes("AS")) * duration.load() / Int(31557600) * App.globalGet(Bytes("FR")) / Int(10000) # I think there might be truncation issues, especially with very high denominator values.
```

Comment:
For the most part, I'm guessing truncation won't be an issue since:
* We only need to calculate rewards once or twice
* The asset stake unit is extremely small so the actual reward growth will outpace the denominator

I think a possible scenario where this might be a problem is when someone keeps calling `calculate_reward`, and then the reward will keep on getting truncated to 0 and they won't get any rewards.